src/App.jsx

```diff
@@ -14,33 +16,82 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function applySort(arr, sortingType, reversed) {
+  const sortedArray = [...arr];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  switch (sortingType) {
+    case 'alphabet':
+      sortedArray.sort((a, b) => a.localeCompare(b));
+      break;
+    case 'length': {
+      sortedArray.sort((a, b) => a.length - b.length);
+      break;
+    }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    default:
+      break;
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (reversed) {
+    sortedArray.reverse();
+  }
+
+  return sortedArray;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortType, setSortType] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const sortGoods = applySort(goodsFromServer, sortType, isReversed);
+  const shouldShowReset = sortType || isReversed;
```

Comment:
The `shouldShowReset` variable is currently set to be true if either `sortType` or `isReversed` is truthy. This means the Reset button will be enabled even if no sorting has been applied but the reverse option is toggled. Consider refining this logic to ensure the Reset button is only enabled when a sort type is selected or the list is reversed.